lib/cc/engine/analyzers/analyzer_base.rb

```diff
@@ -120,6 +123,35 @@ def file_list
         def skip?(_path)
           nil
         end
+
+        def parse(file, request_path)
+          processed_source = ProcessedSource.new(file, request_path)
+
+          SexpBuilder.new(processed_source.ast, file).build
+        rescue => ex
+          if unparsable_file_error?(ex)
+            CC.logger.warn("Skipping #{processed_source.path} due to #{ex.class}")
+            CC.logger.warn("Response status: #{ex.response_status}")
+            CC.logger.debug { "Contents:\n#{processed_source.raw_source}" }
+            CC.logger.debug { "Response:\n#{ex.response_body}" }
+            nil
+          elsif ex.is_a?(CC::Parser::Client::NestingDepthError)
+            CC.logger.warn("Skipping #{processed_source.path} due to #{ex.class}")
+            CC.logger.warn(ex.message)
+            CC.logger.debug { "Contents:\n#{processed_source.raw_source}" }
+            nil
+          else
+            CC.logger.error("Error processing file: #{processed_source.path}")
+            CC.logger.error(ex.message)
+            CC.logger.debug { "Contents:\n#{processed_source.raw_source}" }
+            raise
+          end
+        end
```

Comment:
I see this code is just being moved from one file to the other, but one potential improvement is to define multiple `rescue` statements, rather than `if` and `#is_a?`. For example:

```ruby
def parse(file, request_path)
  // …
rescue CC::Parser::Client::HTTPError => ex
  // …
rescue CC::Parser::Client::NestingDepthError => ex
  // …
else
  // …
end
```